core/src/main/java/org/helsinki/paytrail/request/common/PaytrailPostRequest.java

```diff
@@ -11,17 +11,20 @@
 import org.helsinki.paytrail.response.PaytrailResponse;
 
 import java.nio.charset.StandardCharsets;
+import java.util.Map;
 
 @Slf4j
-public abstract class PaytrailPostRequest<T extends PaytrailResponse, B extends PaytrailPayload<?>> extends PaytrailRequest<T> {
+public abstract class PaytrailPostRequest<T extends PaytrailResponse> extends PaytrailRequest<T> {
 
-	protected abstract B getPayload(PaytrailClient client);
+	protected abstract PaytrailPayload<?> getPayload(PaytrailClient client);
+	protected abstract Map<String, String> getRequestSpecificHeaders();
 
 	@Override
 	public Request formRequest(PaytrailClient client) {
 		Request.Builder req = new Request.Builder().url(this.formUrl(client));
 		req = applyHeaders(client, req);
 		req = applyDefaultAuthHeaders(client, req, String.valueOf(CheckoutMethod.POST));
+		req = applyCustomHeaders(req, getRequestSpecificHeaders());
```

Comment:
Is that header map to be sorted alphabetically? Because these headers must be in correct order. This might not be issue now but can cause some problems later on,

Reply:
Sure, I'll make the change.